src/jinja2/environment.py

```diff
@@ -452,8 +452,9 @@ def getitem(self, obj, argument):
                 else:
                     try:
                         return getattr(obj, attr)
-                    except AttributeError:
-                        pass
+                    except AttributeError as e:
+                        if isinstance(getattr(type(obj), attr, None), property):
```

Comment:
This doesn't work for descriptors in general, only `property`.

Reply:
This also relies on `type(obj)` having the attribute. If it doesn't, it will raise its own error and be even more confusing.